ml-proto/src/script.ml

```diff
@@ -71,6 +75,42 @@ let run_command cmd =
       Error.error cmd.at "assertion failed"
     end
 
+  | AssertHeapEq (offset, expected) ->
+    let m = match !current_module with
+      | Some m -> m
+      | None -> Error.error cmd.at "no module defined to examine heap"
+    in
+    let mem = Eval.get_module_memory m in
+    let count = String.length expected in
+
+    for i=0 to (count-1) do
+      let expected_ch = Char.code expected.[i] in
+      let actual_ch = match Memory.load mem i UInt8Mem Int32Type with
```

Comment:
You probably want to make sure that the index is within bounds (either catch the Bounds exception or by adding a `size` function to the `memory` interface that you can compare against).